paymentsheet/src/main/java/com/stripe/android/paymentsheet/injection/PaymentOptionsViewModelModule.kt

```diff
@@ -0,0 +1,48 @@
+package com.stripe.android.paymentsheet.injection
+
+import android.content.Context
+import com.stripe.android.PaymentConfiguration
+import com.stripe.android.payments.core.injection.IOContext
+import com.stripe.android.payments.core.injection.PUBLISHABLE_KEY
+import com.stripe.android.paymentsheet.DefaultPrefsRepository
+import com.stripe.android.paymentsheet.PaymentSheet
+import com.stripe.android.paymentsheet.PrefsRepository
+import com.stripe.android.paymentsheet.analytics.EventReporter
+import dagger.Module
+import dagger.Provides
+import javax.inject.Named
+import javax.inject.Singleton
+import kotlin.coroutines.CoroutineContext
+
+@Module
+internal class PaymentOptionsViewModelModule {
```

Comment:
Can you remove the dependencies provided here from `FlowControllerModule`, and make this one of the modules of `FlowControllerComponent`?
Then we don't have duplicate places providing `PrefsRepositoryFactory`, `EventReporter.Mode`.

Reply:
Hmm I was trying to do that, but `(PaymentSheet.CustomerConfiguration?) -> PrefsRepository` is not only used for `PaymentOptionsViewModel`, it's also used in [DefaultFlowCongtorllerInitializer](https://github.com/stripe/stripe-android/blob/54276bcccbce9f027a9a7d6eb9d72d666318b7eb/paymentsheet/src/main/java/com/stripe/android/paymentsheet/flowcontroller/DefaultFlowControllerInitializer.kt#L27), so having it provided here introduces some confusion